### PR TITLE
Inherit the run_ordering from DatasetTriggeredTimetable for DatasetOrTimeSchedule

### PR DESCRIPTION
**Pull request by sunank200**
- Inherit the `run_ordering` from `DatasetTriggeredTimetable` for `DatasetOrTimeSchedule`

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by bbovenzi** [comment]
Why do we want to change it on CronTrigger instead of on `DatasetOrTimeSchedule`?

**Comment by sunank200** [comment]
> Why do we want to change it on CronTrigger instead of on `DatasetOrTimeSchedule`?

`DatasetOrTimeSchedule` for example uses the reordering timetable object as per https://github.com/apache/airflow/blob/main/airflow/timetables/datasets.py#L55 For example:
```
schedule=DatasetOrTimeSchedule(
        timetable=CronTriggerTimetable("* * * * *", timezone="UTC"), datasets=(dag1_dataset & dag2_dataset)
    ),
```

**Comment by bbovenzi** [comment]
@uranusjr Does this make sense to you?

**Comment by sunank200** [comment]
> @uranusjr Does this make sense to you?

@bbovenzi I have added the reordering for `DatasetOrTimeSchedule` now.

File: airflow/timetables/datasets.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def __init__(
         self.periodic = timetable.periodic
         self._can_be_scheduled = timetable._can_be_scheduled
 
-        self.run_ordering = timetable.run_ordering
+        self.run_ordering = ("start_date",)
```
Comment (dirrao) [inline airflow/timetables/datasets.py]: If possible, can you add the test case for run ordering?

Comment (uranusjr) [reply airflow/timetables/datasets.py]: Is this choice intentional? `DatasetTriggeredTimetable` (which regular dataset-triggered DAGs use) use `execution_date` instead.

Comment (sunank200) [reply airflow/timetables/datasets.py]: I checked the code for `DatasetTriggeredTimetable` and should inherit the run_order from that class for `DatasetOrTimeSchedule`.

Comment (sunank200) [reply airflow/timetables/datasets.py]: @dirrao added the tests

**Comment by vatsrahul1001** [comment]
@sunank200 tested, LGTM

**Review by uranusjr — APPROVED** [review]

**Comment by dstandish** [comment]
nice job finding the problem @sunank200

**Pull request merged**